client/extensions/woocommerce/state/ui/shipping/zones/locations/selectors.js

```diff
@@ -44,19 +55,30 @@ const getContinentsOwnedByOtherZone = ( state, siteId ) => {
  * @return {Object} A map with the form { countryCode => zoneId }. If a country doesn't appear in the map, it means that
  * it doesn't belong to a zone.
  */
-const getCountriesOwnedByOtherZone = ( state, siteId ) => {
-	const countries = {};
-	const currentZone = getCurrentlyEditingShippingZone( state, siteId );
-	forIn( getRawShippingZoneLocations( state, siteId ), ( { country, postcode }, zoneId ) => {
-		if ( currentZone.id === Number( zoneId ) || ! isEmpty( postcode ) ) {
-			return;
-		}
-		for ( const c of country ) {
-			countries[ c ] = Number( zoneId );
-		}
-	} );
-	return countries;
-};
+const getCountriesOwnedByOtherZone = createSelector(
+	( state, siteId ) => {
+		const countries = {};
+		const currentZone = getCurrentlyEditingShippingZone( state, siteId );
+
+		forIn( getRawShippingZoneLocations( state, siteId ), ( { country, postcode }, zoneId ) => {
+			if ( currentZone.id === Number( zoneId ) || ! isEmpty( postcode ) ) {
+				return;
+			}
+			for ( const c of country ) {
+				countries[ c ] = Number( zoneId );
+			}
+		} );
+		return countries;
+	},
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+		return [
+			siteId,
```

Comment:
`siteId` shouldn't be a dependant on selectors that don't use the `getSelectedSiteId(state)` default argument.